src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -252,6 +252,52 @@ void runtime::intelgpu::do_pad_operation(cldnn::topology& topology,
     topology.add(op_pad);
 }
 
+void runtime::intelgpu::gen_window_loop(codegen::CodeWriter& writer,
+                                        const Shape& output_shape,
+                                        const Shape& win_shape,
+                                        const Shape& win_stride,
+                                        const Shape& pad_below,
+                                        bool is_begin)
+{
+    size_t var_idx = 0;
+
+    if (is_begin)
+    {
+        for (auto const& i : win_shape)
+        {
+            writer << "for (uint w" << var_idx << " = 0; w" << var_idx << " < " << i << "; ++w"
+                   << var_idx << ")\n";
+            writer.block_begin();
+            writer << "const uint win_idx" << var_idx << " = (i" << var_idx + 2 << " * "
+                   << win_stride.at(var_idx) << " /*win_stride*/)"
+                   << " + w" << var_idx << " - " << pad_below.at(var_idx) << " /*pad_below*/;\n";
+            ++var_idx;
+        }
+
+        writer << "if (";
+        // Generate input coordinate condition
+        for (size_t i = 0; i < win_shape.size(); ++i)
+        {
+            if (i)
+            {
+                writer << " && ";
+            }
+            writer << "(win_idx" << i << " >= 0) && (win_idx" << i << " < "
```

Comment:
```
                            if ((win_idx0 >= 0) && (win_idx0 < 4) && (win_idx1 >= 0) && (win_idx1 < 4))
                            {
                                computed_val = input0[i0][i1][i2][i3]/win_elems_size;
                                output[i0][i1][win_idx0][win_idx1] += computed_val;
                            }
```
We don't need to check `(win_idx0 >= 0)` because we use `uint` and the value never be negative.